api/src/main/java/edu/wpi/first/shuffleboard/api/util/TestUtils.java

```diff
@@ -0,0 +1,30 @@
+package edu.wpi.first.shuffleboard.api.util;
+
+/**
+ * Utility class for making sure certain methods may only be called from tests.
+ */
+public final class TestUtils {
+
+  private TestUtils() {
+    throw new UnsupportedOperationException("TestUtils is a utility class!");
+  }
+
+  /**
+   * Asserts that this method's call stack originated from a test method invocation.
+   *
+   * @throws IllegalStateException if this method is not called from within a test
+   */
+  public static void assertRunningFromTest() throws IllegalStateException {
```

Comment:
I searched your changes and this method isn't called anywhere.

If this is really what you wanted wouldn't you do something like `assertNotRunningFromTest`? Even that sounds really bad and should be avoided.

You can apply the dependency inversion principle without using a DI framework. At least to start you can begin by creating constructors that take the objects they depend upon and still have a default no-arg constructor that uses the scope singletons for the time being.

Also, this method won't work when you have a method called on the javaFX thread in a unit test.